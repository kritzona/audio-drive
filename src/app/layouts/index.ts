export { default as DashboardLayout } from './dashboard/DashboardLayout.vue';
