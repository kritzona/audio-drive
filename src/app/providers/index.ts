export { pinia } from './pinia';

export { vuetify } from './vuetify';
export { shallowVuetify } from './shallow-vuetify';

export { router } from './router';
