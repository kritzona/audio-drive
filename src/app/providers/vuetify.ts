import { createVuetify, ThemeDefinition } from 'vuetify';
import 'vuetify/styles';
import * as components from 'vuetify/components';
import * as directives from 'vuetify/directives';
import { aliases, mdi } from 'vuetify/iconsets/mdi';
import { md3 } from 'vuetify/blueprints';

const appTheme: ThemeDefinition = {
  dark: false,
  colors: {
    background: '#FFFFFF',
    surface: '#FFFFFF',
    primary: '#80489C',
    'primary-darken-1': '#432C7A',
    secondary: '#FF8FB1',
    'secondary-darken-1': '#FCE2DB',
    error: '#B00020',
    info: '#2196F3',
    success: '#4CAF50',
    warning: '#FB8C00',
  },
};

export const vuetify = createVuetify({
  components,
  directives,
  icons: {
    defaultSet: 'mdi',
    aliases,
    sets: {
      mdi,
    },
  },
  blueprint: md3,
  theme: {
    defaultTheme: 'appTheme',
    themes: {
      appTheme,
    },
  },
});
