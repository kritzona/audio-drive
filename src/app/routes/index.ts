import { RouteRecordRaw } from 'vue-router';
import { Routes } from '@/shared/constants';

const routes: RouteRecordRaw[] = [
  {
    path: '/',
    name: Routes.MAIN_PAGE,
    component: () => import('@/pages/main/MainPage.vue'),
    beforeEnter: (_to, _from, next) => {
      next({ name: Routes.ALL_PLAYLIST_PAGE });
    },
  },
  {
    path: `/${Routes.UPLOAD_AUDIO_TRACK_PAGE}`,
    name: Routes.UPLOAD_AUDIO_TRACK_PAGE,
    component: () =>
      import('@/pages/upload-audio-track/UploadAudioTrackPage.vue'),
    meta: {
      title: 'Загрузка аудио-трека',
      icon: 'mdi-upload',
    },
  },
  {
    path: `/${Routes.ALL_PLAYLIST_PAGE}`,
    name: Routes.ALL_PLAYLIST_PAGE,
    component: () => import('@/pages/all-playlist/AllPlaylistPage.vue'),
    meta: {
      title: 'Все плейлисты',
      icon: 'mdi-playlist-music',
    },
  },
  {
    path: `/${Routes.ADD_PLAYLIST_PAGE}`,
    name: Routes.ADD_PLAYLIST_PAGE,
    component: () => import('@/pages/add-playlist/AddPlaylistPage.vue'),
    meta: {
      title: 'Добавление нового плейлиста',
      icon: 'mdi-playlist-plus',
    },
  },
];

export default routes;
