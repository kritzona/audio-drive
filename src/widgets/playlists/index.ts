export { default as Playlists } from './ui/Playlists.vue';

export { useUserPlaylistStore } from './model/user-playlist.store';
