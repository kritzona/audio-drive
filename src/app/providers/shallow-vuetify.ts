import { createVuetify } from 'vuetify';
import * as components from 'vuetify/components';
import * as directives from 'vuetify/directives';
import { md3 } from 'vuetify/blueprints';

export const shallowVuetify = createVuetify({
  components,
  directives,
  blueprint: md3,
});
