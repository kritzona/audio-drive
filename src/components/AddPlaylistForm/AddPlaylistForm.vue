<template>
  <v-form ref="form" class="add-playlist-form" @submit.prevent="handleSubmit">
    <v-text-field
      v-model="name"
      :readonly="loading"
      :rules="requiredTextRules"
      clearable
      hide-details="auto"
      label="Название плейлиста"
      color="primary"
      required
    />

    <v-btn
      :loading="loading"
      block
      color="primary"
      size="large"
      type="submit"
      class="mt-5"
    >
      Создать
    </v-btn>
  </v-form>
</template>

<script lang="ts" setup>
import { ref } from 'vue';
import { requiredTextRules } from '@/constants/validation-rules.constants';
import { VForm } from 'vuetify/components';
import { useUserPlaylistStore } from '@/app/stores';
import { useForm } from '@/composables/form.composable';

/**
 * Хранилище списка плейлистов
 */
const userPlaylistStore = useUserPlaylistStore();

/**
 * Бизнес-логика для работы с формой
 */
const { form, loading, validate } = useForm();

/**
 * Наименование плейлиста
 */
const name = ref<string>('');

/**
 * Обработчик отправки формы
 */
const handleSubmit = async () => {
  await validate();

  userPlaylistStore.add({
    id: '1',
    name: name.value,
    tracks: [],
  });
};
</script>
