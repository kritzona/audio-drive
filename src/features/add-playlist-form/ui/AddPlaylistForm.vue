<template>
  <v-form ref="form" class="add-playlist-form" @submit.prevent="handleSubmit">
    <v-text-field
      v-model="name"
      name="name"
      :readonly="loading"
      :rules="requiredTextRules"
      clearable
      hide-details="auto"
      label="Название плейлиста"
      color="primary"
      required
    />

    <v-btn
      :loading="loading"
      block
      color="primary"
      size="large"
      type="submit"
      class="mt-5"
    >
      Создать
    </v-btn>
  </v-form>
</template>

<script lang="ts" setup>
import { ref } from 'vue';
import { requiredTextRules } from '@/shared/constants';
import { VForm } from 'vuetify/components';
import { useUserPlaylistStore } from '@/widgets/playlists';
import { useForm } from '@/shared/model';

/**
 * Хранилище списка плейлистов
 */
const userPlaylistStore = useUserPlaylistStore();

/**
 * Бизнес-логика для работы с формой
 */
const { form, loading, validate } = useForm();

/**
 * Наименование плейлиста
 */
const name = ref<string>('');

/**
 * Обработчик отправки формы
 */
const handleSubmit = async () => {
  await validate();

  userPlaylistStore.add({
    id: '1',
    name: name.value,
    tracks: [],
  });
};

defineExpose({
  form,
  loading,
  name,

  handleSubmit,
});
</script>
