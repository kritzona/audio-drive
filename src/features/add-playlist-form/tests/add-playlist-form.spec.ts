import { describe, it, expect, beforeEach } from 'vitest';
import { mount } from '@vue/test-utils';
import AddPlaylistForm from '../ui/AddPlaylistForm.vue';
import * as providers from '@/app/providers';
import { createTestingPinia } from '@pinia/testing';
import { useUserPlaylistStore } from '@/widgets/playlists';

describe('AddPlaylistForm.vue', () => {
  let vuetify: any;

  beforeEach(() => {
    vuetify = providers.shallowVuetify;
  });

  it('render', () => {
    const wrapper = mount(AddPlaylistForm, {
      global: {
        plugins: [vuetify, createTestingPinia()],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('validation failure', async () => {
    const wrapper = mount(AddPlaylistForm, {
      global: {
        plugins: [vuetify, createTestingPinia()],
      },
    });

    await wrapper.find('input[name="name"]').setValue('');

    await expect(wrapper.vm.handleSubmit()).rejects.toEqual(undefined);
  });

  it('successful validation', async () => {
    const wrapper = mount(AddPlaylistForm, {
      global: {
        plugins: [vuetify, createTestingPinia()],
      },
    });

    const userPlaylistStore = useUserPlaylistStore();

    await wrapper.find('input[name="name"]').setValue('Тестовый плейлист');
    await wrapper.vm.handleSubmit();

    expect(userPlaylistStore.add).toHaveBeenCalledOnce();
  });
});
