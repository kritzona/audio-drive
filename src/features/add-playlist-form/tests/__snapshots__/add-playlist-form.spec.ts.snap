// Vitest Snapshot v1

exports[`AddPlaylistForm.vue > render 1`] = `
"<form class=\\"v-form add-playlist-form\\" novalidate=\\"\\">
  <div class=\\"v-input v-input--horizontal v-input--density-default v-text-field\\">
    <!---->
    <div class=\\"v-input__control\\">
      <div class=\\"v-field v-field--appended v-field--variant-filled v-theme--light\\" role=\\"textbox\\">
        <div class=\\"v-field__overlay\\"></div>
        <div class=\\"v-field__loader\\">
          <div class=\\"v-progress-linear v-theme--light\\" style=\\"height: 0px; --v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
            <!---->
            <div class=\\"v-progress-linear__background bg-primary\\" style=\\"width: 100%;\\"></div>
            <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
              <div class=\\"v-progress-linear__indeterminate\\">
                <div class=\\"v-progress-linear__indeterminate long bg-primary\\"></div>
                <div class=\\"v-progress-linear__indeterminate short bg-primary\\"></div>
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <!---->
        <div class=\\"v-field__field\\" data-no-activator=\\"\\"><label class=\\"v-label v-field-label v-field-label--floating\\" aria-hidden=\\"true\\" for=\\"input-0\\">
            <!---->Название плейлиста
          </label><label class=\\"v-label v-field-label\\" for=\\"input-0\\">
            <!---->Название плейлиста
          </label>
          <!----><input name=\\"name\\" size=\\"1\\" type=\\"text\\" id=\\"input-0\\" required=\\"\\" class=\\"v-field__input\\">
          <!---->
        </div>
        <transition-stub name=\\"expand-x-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
          <div class=\\"v-field__clearable\\" style=\\"display: none;\\"><i class=\\"mdi-close-circle mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable\\" role=\\"button\\" aria-hidden=\\"false\\" aria-label=\\"Clear Название плейлиста\\"></i></div>
        </transition-stub>
        <div class=\\"v-field__append-inner\\">
          <!---->
          <!---->
        </div>
        <div class=\\"v-field__outline\\">
          <!---->
          <!---->
        </div>
      </div>
    </div>
    <!---->
    <!---->
  </div><button type=\\"submit\\" class=\\"v-btn v-btn--block v-btn--elevated v-theme--light bg-primary v-btn--density-default rounded-xl v-btn--size-large v-btn--variant-elevated mt-5\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
    <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"> Создать </span>
    <!---->
    <!---->
  </button>
</form>"
`;
